src/app.jsx

```diff
@@ -91,18 +99,25 @@ class Application extends React.Component {
                     this._asyncRequestContainers = null;
                     this.setState({containersMeta: reply.containers || []});
                     this.state.containersMeta.map((container) => {
-                        utils.varlinkCall(utils.PODMAN, "io.podman.InspectContainer", JSON.parse('{"name":"' + container.id + '"}'))
+                        utils.varlinkCall(utils.PODMAN, "io.podman.InspectContainer", {name: container.id})
+
                                 .then(reply => {
-                                    const temp_containers = this.state.containers;
+                                    let temp_containers = this.state.containers;
                                     temp_containers.push(JSON.parse(reply.container));
                                     this.setState({containers: temp_containers});
                                 })
-                                .catch(ex => console.error("Failed to do InspectImage call:", ex, JSON.stringify(ex)));
+                                .catch(ex => console.error("Failed to do InspectContainer call:", ex, JSON.stringify(ex)));
                     });
-                    this.state.containersMeta.map((container) => {
-                        utils.varlinkCall(utils.PODMAN, "io.podman.GetContainerStats", JSON.parse('{"name":"' + container.id + '"}'))
+                    this.state.containersMeta.filter((ele) => {
+                        if (ele.status === "running") {
+                            return true;
+                        } else {
+                            return false;
+                        }
+                    }).map((container) => {
+                        utils.varlinkCall(utils.PODMAN, "io.podman.GetContainerStats", {name: container.id})
                                 .then(reply => {
-                                    const temp_container_stats = this.state.containersStats;
+                                    let temp_container_stats = this.state.containersStats;
                                     if (reply.container) {
                                         temp_container_stats[container.id] = reply.container;
                                     }
```

Comment:
Same in-place modification of state. Keep in mind that `temp_container_stats = this.state.containersStats` does *not* create a copy, it's just an alias name or reference;  or in C terms, a pointer to the same object.